File: py/src/text/chat/basic.py
import boto3

client = boto3.client(service_name='bedrock-runtime', region_name="us-west-2")
MODEL_ID = 'us.amazon.nova-2-lite-v1:0'


def multi_turn_converse():
    history = []

    turns = [
        'What is the capital of France?',
        'What is it famous for?',
    ]

    for user_text in turns:
        history.append({'role': 'user', 'content': [{'text': user_text}]})

        response = client.converse(
            modelId=MODEL_ID,
            messages=history,
        )

        assistant_message = response['output']['message']
        history.append(assistant_message)

        print(f"User: {user_text}")
        print(f"Assistant: {assistant_message['content'][0]['text']}\n")


multi_turn_converse()

File: ts/src/text/chat/Basic.ts
import {
    BedrockRuntimeClient,
    ConverseCommand,
    Message,
} from '@aws-sdk/client-bedrock-runtime'

const client = new BedrockRuntimeClient({ region: 'us-west-2' })
const MODEL_ID = 'us.amazon.nova-2-lite-v1:0'


async function multiTurnConverse() {
    const history: Message[] = []

    const turns: Array<string> = [
        'What is the capital of France?',
        'What is it famous for?',
    ]

    for (const userText of turns) {
        history.push({ role: 'user', content: [{ text: userText }] })

        const response = await client.send(new ConverseCommand({
            modelId: MODEL_ID,
            messages: history,
        }))

        const assistantMessage = response.output?.message!
        history.push(assistantMessage)

        console.log(`User: ${userText}`)
        console.log(`Assistant: ${assistantMessage.content?.[0]?.text}\n`)
    }
}

multiTurnConverse()